src/main/java/net/sf/jabref/model/entry/Author.java

```diff
@@ -74,7 +74,7 @@ public static String addDotIfAbbreviation(String name) {
 
             if(currentChar == '.') {
                 // A.A. -> A. A.
-                if(i + 1 < name.length() && Character.isUpperCase(name.charAt(i + 1))) {
+                if(((i + 1) < name.length()) && Character.isUpperCase(name.charAt(i + 1))) {
```

Comment:
Think, this is the eclipse setting we agreed about half year ago. We add brackets as much as possible to help readers not familiar with operator precedence to understand the code.